stm/stm.ml

```diff
@@ -65,7 +65,7 @@ end
 
 (* During interactive use we cache more states so that Undoing is fast *)
 let interactive () =
-  if !Flags.ide_slave || !Flags.print_emacs || not !Flags.batch_mode then `Yes
+  if !Flags.ide_slave || not !Flags.batch_mode then `Yes
```

Comment:
Well, in fact if the semantics of `-emacs` is "modify the printer to add tags", indeed it could work well with compile. But that also confirms that the check in `Stm` was wrong.